contracts/src/LightClientV3.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.28;
+
+import { BN254 } from "bn254/BN254.sol";
+import { LightClientV2 } from "./LightClientV2.sol";
+import { IPlonkVerifier } from "./interfaces/IPlonkVerifier.sol";
+import { PlonkVerifierV3 } from "./libraries/PlonkVerifierV3.sol";
+
+/// @title LightClientV3
+/// @notice LightClientV2 with an additional root for gas-efficient state authentication and
+/// @notice improved public input derivation for efficiency and future-proof.
+contract LightClientV3 is LightClientV2 {
+    /// @notice a state value signed by validators as part of the extended light client state
+    uint256 public authRoot;
+    /// @notice Unfortunate re-declaration since V2 mark firstEpoch as private
+    uint64 internal _firstEpoch;
```

Comment:
Why does it need to be public? Instead of re-declaring, we could add a getter function instead?

Reply:
it's not public, it used to be `private`, thus I can't get access (or build reader) in V3. 
am i missing something?

Reply:
Oh I didn't know this. We should use "internal" in the future instead of private.